test/hutch.test.js

```diff
@@ -280,4 +379,67 @@ describe('Hutch', function() {
       });
     });
   });
+
+  it('should clear skip option for exclusive consumer after consumer is rejected', function(complete) {
+
+    this.timeout(20000); // Allow up to 15 seconds for windows
+
+    var config = {
+      connectionString: 'amqp://localhost',
+      retryWait:        100
+    };
+
+    var options = {
+      exchange: {
+        name: 'example.exchange.1',
+        type: 'topic'
+      },
+      queue: {
+        name: 'example.queue',
+        prefetch: 1,
+        durable:  true
+      },
+      publish: {
+        persistent: true,
+        expiration: 86400000
+      },
+      exclusive: true,
+      skipNext: true
+    };
+
+    var messages = [];
+    var consumer1 = function(message, done, fail) { messages.push('c1'); done(); };
+    var consumer2 = function(message, done, fail) { messages.push('c2'); done(); };
+
+    var instance1 = new AMQPHutch();
+    var instance2 = new AMQPHutch();
+
+    instance1.initialise(config);
+    instance2.initialise(config);
+
+    instance1.on('ready', function() {
+      instance1.consume(options, consumer1, function(err) {
+
+        instance2.consume(options, consumer2, function(err) {
+          instance1.close(options.queue.name, function(){
+            // Send a message to consumer 2;
+            instance2.publish(options, "Example Message!", function(err, res){});
+          });
+        });
+
+        // Send a message to consumer 1 then close it.
+        instance1.publish(options, "Example Message!", function(err, res){
+          instance1.close(options.queue.name, function(err){});
+        });
+
+        setTimeout(function(){
+          // We should have only processed one of the files.
+          messages[0].should.equal('c2');
```

Comment:
This checks that the first message processed was by consumer 2, not that only one message was processed.